### DISCUSSION TITLE
Nesting training material

### DISCUSSION THREAD

**Discussion by Cesar-IIUNAM** [NOAA-EMC/WW3#1313] [Q&A]
Dear WW3 community.
One of the objectives of my research is to perform nesting in order to have higher resolution in my study area. I wanted to ask for your support with training material or guidelines to develop the nesting using either multi-grid or real spectral boundary conditions (one-way) methodologies. 

I have found some material (below are attached for your consideration) from WW3 training courses, but I believe they are not complete or perhaps somewhat outdated. Please, maybe you can share with me some recent material for performing two-way and one-way nesting?

Thank you for your support. 
César.
[tutorial_inout.pdf](https://github.com/user-attachments/files/17627188/tutorial_inout.pdf)
[WW3-workshop-exercises-day4-profiling (1).pdf](https://github.com/user-attachments/files/17627191/WW3-workshop-exercises-day4-profiling.1.pdf)

**Comment by mickaelaccensi** [comment]
Dear Cesar,

You can find some up-to-date tutorials here : https://gitlab.ifremer.fr/wave/Training/-/wikis/WW3-2023

Also, please read the WW3 manual, there is an appendix about nesting and a section about regression tests which provided some good examples of one-way and 2-ways nesting.

 Best,

**Reply by Cesar-IIUNAM** [reply]
Thank @mickaelaccensi, for sharing the material. It is very helpful. 

I am trying to set up a one way nested simulation using “ww3_shel”. 
I have successfully run the global model, plus I have configured the outputs boundary points; with this, the "nest.ww3" file has been created.
I have problems to include the nested data in the regional model. When I define the inputs boundary points in “ww3_grid.inp”, I get errors after doing “./ww3_grid”. 
I already checked the thread of comments posted on https://github.com/NOAA-EMC/WW3/discussions/959#discussioncomment-5436315 and https://github.com/NOAA-EMC/WW3/discussions/959#discussioncomment-6808449, and I have found useful answers for my case. But, I still can't fix the error, which I share below. 

 *** WAVEWATCH III ERROR IN W3GRID :
     ERROR IN READING FROM INPUT FILE
     IOSTAT =    0

In “ww3_grid.inp” of the regional model I am setting the inputs boundary points as follows [ww3_grid.txt](https://github.com/user-attachments/files/17654078/ww3_grid.txt) [grid_out.txt](https://github.com/user-attachments/files/17654077/grid_out.txt); at the moment I only want to set the northern boundary as open.

One question, should the output boundary points defined in the “ww3_grid.inp” of the global model be exactly the same as the input boundary points of the regional model? 

Thanks for your help.

César.

**Comment by Cesar-IIUNAM** [comment]
Dear WW3 community, kind regards. 
I would be very grateful for any suggestions on how to fix the error I mentioned above.

Best regards,
Cesar.

**Comment by mickaelaccensi** [comment]
First, please use nml file instead of inp file to avoid problem in the format of the configuration file you are facing
The mask of your regional grid must have active boundaries defined, here you use mask-orig which probably does not contains the active boundaries (see gridgen)

Best

**Comment by Cesar-IIUNAM** [comment]
Hi @mickaelaccensi, Thank you for the recommendation. 
I have reviewed the GRIDEGEN and MULTI-GRID guides; I have tried to run the multi-grid configuration explained in the guide, and I get an error, which I don't quite understand. It looks like it does not read the forcing files. I must say that I have generated all the files like wind.wind, ice.ice, current.current and level.level.
I am not sure if in “ww3_multi.nml” these names should be written like this (e.g. wind.wind., etc). I attach some log.*; 

please can you tell me what is the error?

Thanks for your help,
regards,
Cesar.

[BENG-3M.txt](https://github.com/user-attachments/files/17875321/BENG-3M.txt)
[GLOB-30M.txt](https://github.com/user-attachments/files/17875322/GLOB-30M.txt)
[mww3.txt](https://github.com/user-attachments/files/17875323/mww3.txt)
[ww3_multi.txt](https://github.com/user-attachments/files/17875332/ww3_multi.txt)
[ww3_multi_out.txt](https://github.com/user-attachments/files/17875333/ww3_multi_out.txt)
[ZA-7M.txt](https://github.com/user-attachments/files/17875334/ZA-7M.txt)

**Reply by mickaelaccensi** [reply]
you probably missed a step in the tutorial. The error message is because no active boundaries are found on the grid ZA-7M. So maybe your mod_def is not correct. Please verify that you have active boundaries defined for ZA-7M and BENG-3M when you run ww3_grid. The information is at the end of the respective log files.

**Reply by Cesar-IIUNAM** [reply]
Hi @mickaelaccensi, thanks for replying..

That's true, I had the ZA-3M borders closed. Following the GRIDGEN guide, I have created the bathymetries for the subdomains and defined the borders of those domains as open. 
I have followed the guides carefully, but I am unable to correct the attached error. I am probably missing a step. I am using the auxiliary programs compiled for the cluster, with which I have run other WW3 applications.
I have not modified the “ww3_multi.nml”, I have left it as it comes in the course. 
Consequently, it is not clear to me if the names of the input forcing files are typed only as “wind” or “wind.wind” and so on for the other forcing files?
I attach some outputs of the model in case it can help to identify the error.
Please let me know your opinion. 

The best,
Cesar.

[ww3_multi.nml.txt](https://github.com/user-attachments/files/17962926/ww3_multi.nml.txt)
[job_1734.txt](https://github.com/user-attachments/files/17962927/job_1734.txt)
[mww3_log.txt](https://github.com/user-attachments/files/17962928/mww3_log.txt)
[GLOB-30M_log.txt](https://github.com/user-attachments/files/17962929/GLOB-30M_log.txt)
[BENG-3M_log.txt](https://github.com/user-attachments/files/17962930/BENG-3M_log.txt)
[ww3_multi.txt](https://github.com/user-attachments/files/17962931/ww3_multi.txt)

**Reply by mickaelaccensi** [reply]
in the log, the error message is "'ALLOCATED FAILED". So the issue is probably due to not enough memory available for the job

**Reply by Cesar-IIUNAM** [reply]
Thank you for reviewing the multi-grid outputs.
By memory for the job do you mean cores or storage like hard disk?

**Reply by mickaelaccensi** [reply]
I mean RAM memory to allocate the WW3 array

**Comment by walidchikhi** [comment]
Dear @Cesar-IIUNAM,
I am currently working for the first time on running WW3 over Algeria, using one grid covering the entire Mediterranean and another with higher resolution focused on the Algerian coast. However, I am facing some issues with the nesting. Do you happen to have any  test cases that I could use to reproduce the work over Algeria ? 
Warm regards ,

**Reply by Cesar-IIUNAM** [reply]
Hi @walidchikhi,
I didn't continue working with the multi-grid method, but I made progress with one-way nesting. 
What type of nesting do you want to implement, multi-grid or one-way nesting using “ww3_shel”? 

regards,

**Reply by walidchikhi** [reply]
Hi @Cesar-IIUNAM  , 
I think i will start with the on one way nesting. and i am stuck with the Boundry conditions generation. 
Do you happen to have any material ( Tutorial , test case , documentation... etc) that could help me please ? 
Thanks in advance !
Walid

**Reply by mickaelaccensi** [reply]
Hi Walid,

Please have a look in the WW3 manual, there is a list a regression test with some using spectral boundaries. You can also find tutorials on our wiki page : https://gitlab.ifremer.fr/wave/Training/-/wikis/WW3-2023

Best,
